src/c++/library/http_client.cc

```diff
@@ -1468,7 +1468,12 @@ InferenceServerHttpClient::Infer(
   if (curl_status == CURLE_OPERATION_TIMEDOUT) {
     sync_request->http_code_ = 499;
   } else if (curl_status != CURLE_OK) {
```

Comment:
Question: when the curl_status is not ok - does it make sense to have the same handling. In both cases, return "HTTP Client failed". In both cases I would leave the http_code_ unset (As it's not coming from the server and not a true HTTP error code) and return the error with the error string. That would stop processing. 499 seems to be NGINX specific and 400 seems to indicate there was something wrong with the request.

Reply:
Originally I left the timeout porton intact as it appeared to be handled specially from @tanmayv25 changes a while ago, but upon further inspection it appears to just ultimately return an error, and still prints out the misleading Timer error:
```
root@rmccormick-dt:/mnt/triton/jira/4524-client-growth/client/build# ./install/bin/simple_http_infer_client -t 1000
Failed to update context stat: Timer not set correctly.
error: unable to run model: Deadline Exceeded
```

I think if we return early with an error here, it would have the same behavior but not print any extraneous errors. Sound reasonable @tanmayv25 ?

Reply:
Separately, @tanmayv25 I notice we accept `options.client_timeout` in microseconds, but use `CURLOPT_TIMEOUT_MS` in milliseconds for HTTP client. So when `client_timeout <= 1000`, it [gets rounded down to 0](https://github.com/triton-inference-server/client/blob/a288b4e976c0051e38189be99e29e5eb475c06c3/src/c%2B%2B/library/http_client.cc#L1801-L1804) which means [no timeout from curl's perspective](https://curl.se/libcurl/c/CURLOPT_TIMEOUT_MS.html).

```
root@rmccormick-dt:/mnt/triton/jira/4524-client-growth/client/build# ./install/bin/simple_http_infer_client -t 1
...
PASS : Infer
```

Do we document this anywhere? I was trying to quickly reproduce a timeout by setting it to `1` microsecond, but no timeout occurred because it was rounded down to zero internally. This behavior wasn't clear to me until I read through the code.

Looks like GRPC client [supports microseconds](https://github.com/triton-inference-server/client/blob/a288b4e976c0051e38189be99e29e5eb475c06c3/src/c%2B%2B/library/grpc_client.cc#L1028-L1032) so no issue there:
```
root@rmccormick-dt:/mnt/triton/jira/4524-client-growth/client/build# ./install/bin/simple_grpc_infer_client -t 1   
error: unable to run model: Deadline Exceeded
```

Reply:
I clarified the rounding down issue with timeout in the client_timeout header